plugin/trino-memsql/src/main/java/io/trino/plugin/memsql/MemSqlClient.java

```diff
@@ -624,22 +633,70 @@ protected boolean isSupportedJoinCondition(ConnectorSession session, JdbcJoinCon
         }
 
         String typeName = typeHandle.getJdbcTypeName().get();
-        if (typeName.equalsIgnoreCase("tinyint unsigned")) {
-            return Optional.of(smallintColumnMapping());
-        }
-        if (typeName.equalsIgnoreCase("smallint unsigned")) {
-            return Optional.of(integerColumnMapping());
-        }
-        if (typeName.equalsIgnoreCase("int unsigned")) {
-            return Optional.of(bigintColumnMapping());
-        }
-        if (typeName.equalsIgnoreCase("bigint unsigned")) {
-            return Optional.of(decimalColumnMapping(createDecimalType(20)));
+        if (UNSIGNED_TYPE_REGEX.matcher(typeName).matches()) {
+            switch (typeHandle.getJdbcType()) {
+                case Types.BIT:
+                    return Optional.of(booleanColumnMapping());
+
+                case Types.TINYINT:
+                    return Optional.of(smallintColumnMapping());
+
+                case Types.SMALLINT:
+                    return Optional.of(integerColumnMapping());
+
+                case Types.INTEGER:
+                    return Optional.of(bigintColumnMapping());
+
+                case Types.BIGINT:
+                    return Optional.of(decimalColumnMapping(createDecimalType(20)));
+            }
         }
 
         return Optional.empty();
     }
 
+    private static ColumnMapping memSqlTimeColumnMapping(TimeType timeType)
+    {
+        return ColumnMapping.longMapping(
+                timeType,
+                memsqlTimeReadFunction(timeType),
+                timeWriteFunction(timeType.getPrecision()));
+    }
+
+    private static LongReadFunction memsqlTimeReadFunction(TimeType timeType)
+    {
+        requireNonNull(timeType, "timeType is null");
+        checkArgument(timeType.getPrecision() <= 9, "Unsupported type precision: %s", timeType);
+        int precision = timeType.getPrecision();
+        return (resultSet, columnIndex) -> {
+            // SingleStore driver wraps values <00:00:00 and >24:00:00, but we must disallow them
+            // LocalTime time = resultSet.getObject(columnIndex, LocalTime.class);
```

Comment:
Doesn't the commented code already do that? Since `LocalTime` cannot represent values outside the range anyway? Why the change?

Reply:
Discussed offline. The driver is too smart and wraps values > 24h or <0h to be within the 0-24 range which is "incorrect". So to preserve semantics we need to read as string and apply verification on our end.